segment-events-correlation-issues-analysis.ipynb

```diff
@@ -0,0 +1,286 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Event correlation issues analysis (Analytics/Schema2 Telemetry)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Synchronize all folders from S3 bucket into the data/ dir.\n",
+    "\n",
+    "Example:\n",
+    "\n",
+    "- data/\n",
+    "  - 1709164800000/\n",
+    "    - gzip files\n",
+    "  - 1709251200000/\n",
+    "    - gzip files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### List all folders inside data/"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "dataframes_root_path = \"data\"\n",
+    "\n",
+    "files_dir = [\n",
+    "    f for f in os.listdir(dataframes_root_path) if os.path.isdir(os.path.join(dataframes_root_path, f))\n",
+    "]\n",
+    "print(files_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate all.jsonl files (per folder & global)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "cmd = f\"rm -f ./data/all.jsonl\"\n",
```

Comment:
@romartin apologies for the delayed review and thanks for the contribution!

Minor comment: typically it's a good idea to avoid deleting or overwriting files in scripts, instead writing to a new unique file and letting the user clean up the old ones so they'll have their old files in case of script failure.

Second best approach is overwriting only after the script completes successfully; write all output to `data/all.jsonl.$$` and then `mv data/all.jsonl.$$ data/all.jsonl` if there aren't any errors generating `data/all.jsonl.$$`